### PR TITLE
Update integration to new installation methods.

### PR DESCRIPTION
**Pull request by pszczesniak**
## Description

This PR should update integration to new installation methods of `CKEditor5`.

## Steps to reproduce

It's improving `CKEditor5` installation methods. See https://github.com/ckeditor/ckeditor5/issues/15502

---

[#taskid X](https://wiris.kanbanize.com/ctrl_board/2/cards/X/details/) (_Associated Kanbanize card, if any_)

### CODE REVIEW

File: packages/ckeditor5/package.json [hunk 1]
```diff
@@ -25,20 +25,35 @@
...
-    "@ckeditor/ckeditor5-core": ">=27.0.0",
-    "@ckeditor/ckeditor5-engine": ">=27.0.0",
-    "@ckeditor/ckeditor5-ui": ">=27.0.0",
-    "@ckeditor/ckeditor5-widget": ">=27.0.0",
-    "@wiris/mathtype-html-integration-devkit": "1.17.1"
+    "@wiris/mathtype-html-integration-devkit": "1.17.2"
+  },
+  "devDependencies": {
+    "@ckeditor/ckeditor5-dev-build-tools": "^40.2.0",
+    "ckeditor5": "nightly"
```
Comment (pszczesniak) [inline packages/ckeditor5/package.json:54]: After the release the `CKEditor5` in version 42 it should be changed from nightly to the latest version (`>=42.0.0`).

**Review by carla-at-wiris — APPROVED** [review]

**Comment by carla-at-wiris** [comment]
To validate our example demos, HTML and Vue, work as expected with the new installation method, build and deploy them as follows:

`nx build ckeditor5`

once finished run:
`nx start html-ckeditor5` or `nx start vue-ckeditor5`

**Comment by xjiang-at-wiris** [comment]
`error cssnano@7.0.2: The engine "node" is incompatible with this module. Expected version "^18.12.0 || ^20.9.0 || >=22.0". Got "21.7.3"`

Note that after updating dependency, Node 21 will no longer be usable (I was previously using version 21.7.3).

**Comment by xjiang-at-wiris** [comment]
> To validate our example demos, HTML and Vue, work as expected with the new installation method, build and deploy them as follows:
> 
> `nx build ckeditor5`
> 
> once finished run: `nx start html-ckeditor5` or `nx start vue-ckeditor5`


None of this two commands works for me... (after changing node to v20.12.2, having `yarn && nx build ckeditor5` executed)

```
> nx run html-ckeditor5:start

<i> [webpack-dev-server] Project is running at:
<i> [webpack-dev-server] Loopback: http://localhost:8002/
<i> [webpack-dev-server] On Your Network (IPv4): http://192.168.1.122:8002/
<i> [webpack-dev-server] Content not from webpack is served from '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/' directory

 NX  Web Development Server is listening at http://0.0.0.0:8002/

node:internal/fs/watchers:247
    const error = new UVException({
                  ^

Error: ENOSPC: System limit for number of file watchers reached, watch '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/'
    at FSWatcher.<computed> (node:internal/fs/watchers:247:19)
    at Object.watch (node:fs:2464:36)
    at createFsWatchInstance (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:119:15)
    at setFsWatchListener (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:166:15)
    at NodeFsHandler._watchWithNodeFs (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:331:14)
    at NodeFsHandler._handleDir (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:567:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async NodeFsHandler._addToNodeFs (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:617:16)
    at async /home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/index.js:451:21
    at async Promise.all (index 0)
Emitted 'error' event on FSWatcher instance at:
    at FSWatcher._handleError (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/index.js:647:10)
    at NodeFsHandler._addToNodeFs (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:645:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/index.js:451:21
    at async Promise.all (index 0) {
  errno: -28,
  syscall: 'watch',
  code: 'ENOSPC',
  path: '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/',
  filename: '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/'
}

Node.js v20.12.2

 NX   Running target start for project html-ckeditor5 and 1 task it depends on failed

Failed tasks:

- html-ckeditor5:start

Hint: run the command with --verbose for more details.

```

**Comment by psala-at-wiris** [comment]
> > To validate our example demos, HTML and Vue, work as expected with the new installation method, build and deploy them as follows:
> > `nx build ckeditor5`
> > once finished run: `nx start html-ckeditor5` or `nx start vue-ckeditor5`
> 
> None of this two commands works for me... (after changing node to v20.12.2, having `yarn && nx build ckeditor5` executed)
> 
> ```
> > nx run html-ckeditor5:start
> 
> <i> [webpack-dev-server] Project is running at:
> <i> [webpack-dev-server] Loopback: http://localhost:8002/
> <i> [webpack-dev-server] On Your Network (IPv4): http://192.168.1.122:8002/
> <i> [webpack-dev-server] Content not from webpack is served from '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/' directory
> 
>  NX  Web Development Server is listening at http://0.0.0.0:8002/
> 
> node:internal/fs/watchers:247
>     const error = new UVException({
>                   ^
> 
> Error: ENOSPC: System limit for number of file watchers reached, watch '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/'
>     at FSWatcher.<computed> (node:internal/fs/watchers:247:19)
>     at Object.watch (node:fs:2464:36)
>     at createFsWatchInstance (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:119:15)
>     at setFsWatchListener (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:166:15)
>     at NodeFsHandler._watchWithNodeFs (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:331:14)
>     at NodeFsHandler._handleDir (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:567:19)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async NodeFsHandler._addToNodeFs (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:617:16)
>     at async /home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/index.js:451:21
>     at async Promise.all (index 0)
> Emitted 'error' event on FSWatcher instance at:
>     at FSWatcher._handleError (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/index.js:647:10)
>     at NodeFsHandler._addToNodeFs (/home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/lib/nodefs-handler.js:645:18)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async /home/xinyujiangwiriscom/Descargas/html-integrations/node_modules/chokidar/index.js:451:21
>     at async Promise.all (index 0) {
>   errno: -28,
>   syscall: 'watch',
>   code: 'ENOSPC',
>   path: '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/',
>   filename: '/home/xinyujiangwiriscom/Descargas/html-integrations/demos/html/ckeditor5/'
> }
> 
> Node.js v20.12.2
> 
>  NX   Running target start for project html-ckeditor5 and 1 task it depends on failed
> 
> Failed tasks:
> 
> - html-ckeditor5:start
> 
> Hint: run the command with --verbose for more details.
> ```

Hi @xjiang-at-wiris,

This is probably related to your system's inotify limits. On Ubuntu, you can check this limit with the following command:
- `grep fs.inotify.max_user_watches /etc/sysctl.conf`
  
If the limit is set or close to 8192 -the defaul value in Ubuntu- it might not be enough for large projects.   
[More info](https://watchexec.github.io/docs/inotify-limits.html#:~:text=The%20default%20max_user_watches%20limit%20on,might%20not%20be%20aware%20of!)

**Review by psala-at-wiris — DISMISSED** [review]

**Comment by xjiang-at-wiris** [comment]
> nx start html-ckeditor5

It works, thank you!

**Review by usantos-at-wiris — COMMENTED** [review]
With this changes comes afloat a new warning regarding CK5 versions. Consider watching!  Great job!🎉

````json
warning Pattern ["ckeditor5@nightly"] is trying to unpack in the same destination "/home/ubaldo-wiris/.cache/yarn/v6/npm-ckeditor5-0.0.0-nightly-20240614.0-37168ff953eda3c113019aea5028efc7d652e20d-integrity/node_modules/ckeditor5" as pattern ["ckeditor5@>=42.0.0 || ^0.0.0-nightly","ckeditor5@>=42.0.0 || ^0.0.0-nightly","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0","ckeditor5@0.0.0-nightly-20240614.0"]. This could result in non-deterministic behavior, skipping.

**Review by xjiang-at-wiris — APPROVED** [review]

**Review by carla-at-wiris — APPROVED** [review]

**Comment by carla-at-wiris** [comment]
Commits merged to master using cherry-pick

**Pull request closed without merging**